roofit/roofitcore/src/RooTreeDataStore.cxx

```diff
@@ -1365,11 +1335,52 @@ void RooTreeDataStore::Draw(Option_t* option)
 
 void RooTreeDataStore::Streamer(TBuffer &R__b)
 {
-   if (R__b.IsReading()) {
-      R__b.ReadClassBuffer(RooTreeDataStore::Class(),this);
-      initialize() ;
-   } else {
-      R__b.WriteClassBuffer(RooTreeDataStore::Class(),this);
-   }
+  if (R__b.IsReading()) {
+    UInt_t R__s, R__c;
+    const Version_t R__v = R__b.ReadVersion(&R__s, &R__c);
+
+    R__b.ReadClassBuffer(RooTreeDataStore::Class(), this, R__v, R__s, R__c);
+
+    if (!_tree) {
+      TFile* parent = dynamic_cast<TFile*>(R__b.GetParent());
+      assert(parent);
+      parent->GetObject(makeTreeName().c_str(), _tree);
+//      if (_tree) {
```

Comment:
Rather than keeping the comment-out code, it might be worth explaining in text the semantic difference (i.e. expand on "let's keep the tree attached to the file so it can retrieve the data")